wordpress-mobile/WordPress-Android
# [Tooling] Remove release_notes.xml file now that changelog upload has been automated via supply

Back in https://github.com/wordpress-mobile/WordPress-Android/pull/16235, I implemented the automation of uploading the `changelogs/*.txt` release notes to the PlayStore using fastlane's `upload_to_play_store` (aka `supply`)

While this seemed to have worked great, so far I kept the `android_create_xml_release_notes` action around a little longer just in case I would notice issues with the new changelog upload automation.

This PR prepares the removal of the remaining traces of `android_create_xml_release_notes` (and the `release_notes.xml` files it created) so that we can land it when we're ready.

ℹ️  I'm marking it with the `Not Ready for Merge` label only because we might want to wait to see if the submission of 19.6 worked before merging. (Though honestly if we prefer to land it before, we could always revert in case some errors with the new automation start to appear and we need the XML it back until we fix them…)

## To test

Nothing to test on the PR per se. But cc-ing @oguzkocer as you'll cover for me for the `finalize_release` of 19.6 and the code freeze of 19.7:
 - Once you you'll run `finalize_release` for 19.6 then go to the PlayStore to submit it to Google for review, don't hesitate to double-check that the automation properly updated the release notes for all locales in the PlayStore, with the correct copies for 19.6, and without you having to paste them manually (nor fix any issue manually on those notes)
 - Then if that worked correctly, feel free to remove the `[Status] Not Ready for Merge` label, then merge this PR in `trunk` before doing the code-freeze for `19.7` (or retarget this PR to `release/19.7` after the code freeze otherwise).
